src/components/entries/FEEL/FeelPopup.js

```diff
@@ -54,11 +66,48 @@ export default function FEELPopupRoot(props) {
 
   // close popup on element change, cf. https://github.com/bpmn-io/properties-panel/issues/270
   useEffect(() => {
-    if (element && element !== prevElement) {
+    if (element && prevElement && element !== prevElement) {
       handleClose();
     }
   }, [ element ]);
 
+  // allow close and open via events
+  useEffect(() => {
+
+    if (!eventBus) {
+      return;
+    }
+
+    const handlePopupOpen = (context) => {
+      const {
+        entryId,
+        popupConfig,
+        sourceElement
+      } = context;
+
+      handleOpen(entryId, popupConfig, sourceElement);
+    };
+
+    const handleIsOpen = () => {
+      return isOpen();
+    };
+
+    eventBus.on('feelPopup.close', handleClose);
+    eventBus.on('feelPopup.open', handlePopupOpen);
+    eventBus.on('feelPopup.isOpen', handleIsOpen);
+
+    return () => {
+      if (!eventBus) {
+        return;
+      }
+
+      eventBus.off('feelPopup.close', handleClose);
```

Comment:
I suggest that we separate external events you're supposed to use (`feelPopup.opened`) from internal ones (i.e. `_open`, `_isOpen`, ...).

Other than that this is a great addition. Shows how we can make core react components accessible to the outside, behind a clear, known, contract. :muscle:

Reply:
Good suggestion, makes a lot of sense! Updated it accordingly 👍